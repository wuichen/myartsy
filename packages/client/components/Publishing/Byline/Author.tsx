import { Sans, SansSize } from "@artsy/palette"
import React from "react"
import styled from "styled-components"
import { getAuthorByline } from "../Constants"
import { BylineLayout } from "../Typings"

interface AuthorProps {
  authors?: any
  color?: string
  layout?: BylineLayout
  size?: SansSize
}

export const Author: React.SFC<AuthorProps> = props => {
  const { color, layout, size } = props
  const condensed = layout === "condensed"
  const fontSize = size ? size : condensed ? "2" : "3t"

  return (
    <Sans size={fontSize} weight="medium">
      <StyledAuthor condensed={condensed} color={color}>
        {getAuthorByline(props.authors)}
      </StyledAuthor>
    </Sans>
  )
}

Author.defaultProps = {
  color: "black",
}

export const StyledAuthor = styled.div.attrs<{
  condensed?: boolean
  color: string
}>({})`
  margin: 10px 20px 0 0;
  ${props => bullet(props.color, props.condensed)};
`

const bullet = (color, condensed = false) => {
  return `
    &::before {
      content: "";
      display: inline-block;
      width: 8px;
      height: 8px;
      border-radius: 50%;
      margin: ${condensed ? "0 5px 1px 0;" : "5px 5px 1px 0"};
      background-color: ${color};
    }
  `
}
