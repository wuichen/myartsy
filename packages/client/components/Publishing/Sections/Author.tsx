import { Box, Flex, Sans, space } from "@artsy/palette"
import { pMedia } from "Components/Helpers"
import Icon from "Components/Icon"
import React from "react"
import Markdown from "react-markdown"
import styled from "styled-components"
import { resize } from "Utils/resizer"

interface AuthorProps {
  author: any
  color?: string
}

export const Author: React.SFC<AuthorProps> = ({ author, color = "black" }) => {
  const profileImage = author.image_url ? (
    <ProfileImage mr={20} src={resize(author.image_url, { width: 200 })} />
  ) : (
    false
  )
  return (
    <AuthorContainer color={color} mb={20} alignItems="center">
      {profileImage}
      <Sans size="4" weight="medium">
        <AuthorInfo>
          {author.bio && author.bio.length ? (
            <Markdown
              source={author.bio}
              disallowedTypes={["Paragraph"]}
              unwrapDisallowed
              containerTagName="span"
            />
          ) : (
            <div>{author.name}</div>
          )}
        </AuthorInfo>

        {author.twitter_handle && author.twitter_handle.length ? (
          <span>
            <TwitterHandle href={`http://twitter.com/${author.twitter_handle}`}>
              <Icon name="twitter" color={color} />
              {`@${author.twitter_handle}`}
            </TwitterHandle>
          </span>
        ) : (
          false
        )}
      </Sans>
    </AuthorContainer>
  )
}

const ProfileImage = styled(Box)<{ src?: string }>`
  min-width: 60px;
  min-height: 60px;
  border-radius: 50%;
  background: url(${props => props.src || ""}) no-repeat center center;
  background-size: cover;

  ${pMedia.xs`
    min-width: 40px;
    min-height: 40px;
  `};
`

const AuthorContainer = styled(Flex)<{ color: string }>`
  color: ${props => props.color};

  a {
    color: ${props => props.color};
  }
`

const AuthorInfo = styled.span`
  margin-right: ${space(2)}px;
`

const TwitterHandle = styled.a`
  text-decoration: none;
  white-space: nowrap;

  ${Icon} {
    vertical-align: middle;
    margin: 0;
  }
`
