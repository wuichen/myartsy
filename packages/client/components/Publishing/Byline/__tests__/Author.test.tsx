import { mount } from "enzyme"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { Author } from "../Author"

describe("Author", () => {
  const author = [{ name: "Molly Gottschalk" }]
  const authors = [
    { name: "Molly Gottschalk" },
    { name: "Casey Lesser" },
    { name: "Anna Louie Sussman" },
  ]

  describe("Snapshots", () => {
    it("Single author", () => {
      renderer.create(<Author authors={author} layout={"split"} />)
      expect(author).toMatchSnapshot()
    })

    it("Multiple authors", () => {
      renderer.create(<Author authors={authors} layout={"split"} />)
      expect(authors).toMatchSnapshot()
    })
  })

  describe("Unit", () => {
    it("renders a single author", () => {
      const component = mount(<Author authors={authors} layout={"split"} />)
      expect(component.text()).toMatch("Molly Gottschalk")
    })

    it("renders multiple authors", () => {
      const component = mount(<Author authors={authors} layout={"split"} />)

      expect(component.text()).toMatch("Molly Gottschalk")
      expect(component.text()).toMatch("Casey Lesser")
      expect(component.text()).toMatch("Anna Louie Sussman")
    })
  })
})
