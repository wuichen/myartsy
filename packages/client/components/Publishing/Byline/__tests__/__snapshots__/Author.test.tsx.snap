// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author Snapshots Multiple authors 1`] = `
Array [
  Object {
    "name": "Molly Gottschalk",
  },
  Object {
    "name": "Casey Lesser",
  },
  Object {
    "name": "Anna Louie Sussman",
  },
]
`;

exports[`Author Snapshots Single author 1`] = `
Array [
  Object {
    "name": "Molly Gottschalk",
  },
]
`;
